<launch>

  <arg name="resolution_width_duo" default="320" />
  <arg name="resolution_height_duo" default="240" />
  <arg name="im_noise" default="5" />
  <arg name="acc_noise" default="1" />
  <arg name="gyro_noise" default="10" />
  <arg name="orientation_noise" default="1" />
  <arg name="pressure_noise" default="1" />
  <arg name="sigma_init" default="0.0001" />
  <arg name="Kp_pos" default="1" />
  <arg name="Kd_pos" default="0" />
  <arg name="Kp_yaw" default="0" />

  <arg name="camera_name" default="101" /> <!-- the camera name is equal to the last 3 digits of the odroid's IP -->
  <arg name="lense_type" default="FS3028B13M8" />

  <arg name="show_tracker_images" default="1" />
  <arg name="debug_publish_freq" default="10" />

  <node name="vio_ros_node" pkg="vio_ros" type="vio_ros_node" output="screen" >
    <remap from="/imu" to="/duo3d_camera/cam_imu"/>
    <remap from="/left_image" to="/duo3d_camera/left/image_raw"/>
    <remap from="/right_image" to="/duo3d_camera/right/image_raw"/>

    <param name="acc_noise" type="double" value="$(arg acc_noise)" />
    <param name="gyro_noise" type="double" value="$(arg gyro_noise)" />
    <param name="process_noise_3" type="double" value="0.0" />

    <param name="im_noise" type="double" value="$(arg im_noise)" />

    <param name="orientation_noise" type="double" value="$(arg orientation_noise)" />
    <param name="pressure_noise" type="double" value="$(arg pressure_noise)" />
    <param name="sigma_init" type="double" value="$(arg sigma_init)" />

    <param name="Kp_pos" type="double" value="$(arg Kp_pos)" />
    <param name="Kd_pos" type="double" value="$(arg Kd_pos)" />
    <param name="Kp_yaw" type="double" value="$(arg Kp_yaw)" />

    <!-- note that these parameters are not free. The total number of points num_anchors*num_points_per_anchor must equal the total number of points with wich SLAM was codered in Matlab. -->
    <param name="num_points_per_anchor" type="int" value="4" />
    <param name="num_anchors" type="int" value="4" />

    <param name="show_tracker_images" type="bool" value="$(arg show_tracker_images)" />
   
    <param name="camera_name" type="string" value="$(arg camera_name)" />
    <param name="lense_type" type="string" value="$(arg lense_type)" />
    <param name="resolution_width" type="int" value="$(arg resolution_width_duo)" />
    <param name="resolution_height" type="int" value="$(arg resolution_height_duo)" />

    <param name="debug_publish_freq" type="double" value="$(arg debug_publish_freq)" />
  </node>

</launch>
